src/graphql/mutations/start_game.ts

```diff
@@ -184,6 +197,61 @@ async function setupGame(
   game.phase = 'movement';
   game.turnGamePlayerId = turnPlayer.id;
   game.status = 'inProgress';
+
+  // Prove gave via the proving service
+  try {
+    await proveInitGame(
+      game,
+      playerOnePieces,
+      playerTwoPieces,
+      turnPlayer,
+      otherPlayer,
+      turnPlayerNumber
+    );
+  } catch (e) {
+    // log and move on with saving state
+    console.log('$$$ Prover Error $$$ ', e);
+  }
+
   await game.save({ transaction: t });
   return game;
 }
+
+async function proveInitGame(
+  game: Models.Game,
+  player1Pieces: Array<Models.GamePiece>,
+  player2Pieces: Array<Models.GamePiece>,
+  player1: Models.GamePlayer,
+  player2: Models.GamePlayer,
+  turnPlayerNumber: number
+) {
+  const piecesInput = [...player1Pieces].concat([...player2Pieces]);
+  const pieces = await serializePiecesTreeFromPieces(piecesInput);
+  const arena = await serializeArenaTreeFromPieces(piecesInput);
+  const p1 = await player1.player();
+  const p2 = await player2.player();
+  const turnsNonce = this.turnNumber || 0;
```

Comment:
Don't know if it matters in this context but the turn number of the first turn will be `1`.

Reply:
I should set this to be 0 always, since it is a new game.

Reply:
Ah I realized this is in the game model, not in the start game mutation.  So it's behaving as expected.  In my testing it's being set to 0.